### ISSUE TITLE
Only the first chunk of email attachments gets uploaded (v2.2.0 and higher)

### ISSUE THREAD

**Issue by niczon** [microsoftgraph/msgraph-sdk-java#529]
There is a problem when uploading an email attachment that is larger than the chunk size.

### Expected behavior

The attachment gets fully uploaded and added to the email message.

### Actual behavior

After uploading the first chunk, we get a success callback and the upload session gets terminated. The attachment does not get added to the email.

![ChunkedUploadResponseHandler](https://user-images.githubusercontent.com/71315274/95201546-66945d00-07e0-11eb-9d69-ecc93e17fc73.PNG)

### Steps to reproduce the behavior

Upload an email attachment larger than the chunk size (5 MB or custom size).

**Comment by baywet** [comment]
Hi @niczon 
Thanks for reaching out.
Can you share the code you have that reproduces the bug?
Are you trying against an Office 365 account or a personal account?

We do have a unit test testing this scenario which passes at the time of writing.
https://github.com/microsoftgraph/msgraph-sdk-java/blob/dev/src/test/java/com/microsoft/graph/functional/OutlookTests.java#L286

**Comment by niczon** [comment]
Hi @baywet 

The code we use is:

```
		AttachmentItem attachmentItem = new AttachmentItem();
		attachmentItem.attachmentType = AttachmentType.FILE;
		attachmentItem.name = file.getName();
		attachmentItem.size = streamSize;

		IProgressCallback<AttachmentItem> callback = new IProgressCallback<AttachmentItem>() {

			@Override
		    public void progress(final long current, final long max) {
		    	_lognode.trace(
		            String.format("Uploaded %d bytes of %d total bytes", current, max)
		        );
		    }

		    @Override
		    public void success(final AttachmentItem result) {
		    	_lognode.debug(
		            String.format("Uploaded attachment: %s", attachmentItem.name)
		        );
		    	success = true;
		    };

		    public void failure(final ClientException ex) {
		    	_lognode.error(
		            String.format("Error uploading attachment: %s", ex.getMessage())
		        );
		    }
		};

		UploadSession uploadSession = graphClient
		    .users(userId)
		    .messages(messageId)
		    .attachments()
		    .createUploadSession(attachmentItem)
		    .buildRequest()
		    .post();

		ChunkedUploadProvider<AttachmentItem> chunkUploadProvider =
		    new ChunkedUploadProvider<AttachmentItem>
		        (uploadSession, graphClient, fileStream, streamSize, AttachmentItem.class);

		int[] customConfig = { (int) (chunkSizeMultiplier * 320 * 1024) };

		chunkUploadProvider.upload(callback, customConfig);
```

Note: for the chunkSizeMultiplier, we currently use 16, which leads to the default chunk size of 5 MB.

The account is an enterprise Office 365 account.

For the upload session, the Outlook API seems to be used, which uses PascalCase instead of camelCase. As you can see in my screenshot, we get a NextExpectedRanges instead of nextExpectedRanges in the response JSON. During deserialization, this gets stored in the AdditionalDataManager of the UploadSession object instead of in the nextExpectedRanges attribute, which causes the issue.

Your unit test will not detect the problem, because it only asserts that the result in the success callback is not null, which is true because the response handler returns an object.

I think you should either take the different capitalization into account, or use the HTTP response codes to determine whether the upload session has been completed or a chunk has been upload.
Chunk completed = HTTP 202 (ODSP) or HTTP 200 (Outlook).
Upload session completed = HTTP 201 (both ODSP and Outlook).

**Comment by baywet** [comment]
thanks for the additional details! I purposely removed dependencies on status codes as services don't behave exactly the same way depending on whether you're uploading to OneDrive for business, OneDrive consumer, SharePoint Online, Exchange Online, or Outlook online (consumer). The behavior is also impacted by the size of the file and other options.

The issue of having the service replying with pascal case instead of dromedary case is not limited to this set of APIs, so I've decided to fix this instead.
I've authored #531 which will close this issue when it gets merged.
